wallet/src/de/schildbach/wallet/ui/explore/ExploreFragment.kt

```diff
@@ -21,64 +21,62 @@ import android.content.Intent
 import de.schildbach.wallet.ui.main.MainViewModel
 import de.schildbach.wallet.ui.staking.StakingActivity
+import de.schildbach.wallet_test.R
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.launch
+import org.dash.wallet.common.Constants
 import org.dash.wallet.common.services.analytics.AnalyticsConstants
-import org.dash.wallet.common.services.analytics.AnalyticsService
 import org.dash.wallet.common.ui.dialogs.AdaptiveDialog
 import org.dash.wallet.common.ui.viewBinding
-import org.dash.wallet.features.exploredash.R
+import org.dash.wallet.common.util.safeNavigate
 import org.dash.wallet.features.exploredash.databinding.FragmentExploreBinding
 import org.dash.wallet.features.exploredash.ui.ExploreTopic
 import java.util.*
-import javax.inject.Inject
 
 @AndroidEntryPoint
 @FlowPreview
 @ExperimentalCoroutinesApi
 class ExploreFragment : Fragment(R.layout.fragment_explore) {
-    private val viewModel by activityViewModels<MainViewModel>()
     private val binding by viewBinding(FragmentExploreBinding::bind)
-    @Inject
-    lateinit var blockChainDao: BlockchainStateDao
-    @Inject
-    lateinit var analytics: AnalyticsService
+    private val viewModel: MainViewModel by activityViewModels()
+
+    private val stakingLauncher = registerForActivityResult(
+        ActivityResultContracts.StartActivityForResult()
+    ) { result ->
+        if (result.resultCode == Constants.USER_BUY_SELL_DASH) {
+            safeNavigate(ExploreFragmentDirections.exploreToBuySell())
+        }
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        enterTransition = MaterialFadeThrough()
 
         binding.merchantsBtn.setOnClickListener {
-            val intent = ExploreActivity.createIntent(requireContext(), ExploreTopic.Merchants)
-            startActivity(intent)
+            viewModel.logEvent(AnalyticsConstants.Explore.WHERE_TO_SPEND)
+            safeNavigate(ExploreFragmentDirections.exploreToSearch(ExploreTopic.Merchants))
         }
 
         binding.atmsBtn.setOnClickListener {
-            val intent = ExploreActivity.createIntent(requireContext(), ExploreTopic.ATMs)
-            startActivity(intent)
+            viewModel.logEvent(AnalyticsConstants.Explore.PORTAL_ATM)
+            safeNavigate(ExploreFragmentDirections.exploreToSearch(ExploreTopic.ATMs))
         }
 
         binding.stakingBtn.setOnClickListener {
-            analytics.logEvent(AnalyticsConstants.CrowdNode.STAKING_ENTRY, bundleOf())
+            viewModel.logEvent(AnalyticsConstants.CrowdNode.STAKING_ENTRY)
             handleStakingNavigation()
         }
 
-        binding.faucetBtn.setOnClickListener {
-            val intent = ExploreActivity.createIntent(requireContext(), ExploreTopic.Faucet)
-            startActivity(intent)
-        }
-
-        setAPY(0.0) // hide the APY
-
```

Comment:
This was removed, but we need the Faucet button functionality.

Reply:
Good catch, fixed